### PR TITLE
2022.1.x dune versions

### PR DESCRIPTION
**Pull request by renefritze**
- [tests] ensure we get no dune import warnings in our CI
- [config] bump supported dune versions

### CODE REVIEW

**Comment by renefritze** [comment]
See also #1709 
@HenKlei saw that our release/main docs carry import warnings on all tutorials. These PRs will prevent that in the future.

**Comment by renefritze** [comment]
We should merge this also before releasing 2022.1.1 via #1724

**Comment by ftschindler** [comment]
> Mirrors #1709

Any reason you're not just cherry-picking the work you did there?

**Comment by renefritze** [comment]
> > Mirrors #1709
> 
> Any reason you're not just cherry-picking the work you did there?

I though I did?

**Comment by ftschindler** [comment]
> > > Mirrors #1709
> > 
> > 
> > Any reason you're not just cherry-picking the work you did there?
> 
> I thought I did?

I was thinking about 85882ace2d88ba10a7b1cedb96250e30c4b42885. That's not required here?

**Comment by renefritze** [comment]
Ah, yeah, we didn't really need it here in the release branch. But I've added it now anyhow, since I wanted to push the CI fix in the branch too and it doesn't hurt any.

**Review by ftschindler — APPROVED** [review]

File: conftest.py [hunk 1]
```diff
@@ -16,11 +16,11 @@
     "verbosity": Verbosity.normal,
 }
 settings.register_profile("ci_large", max_examples=400, **_common_settings)
-settings.register_profile("ci_pr", max_examples=80, **_common_settings)
-settings.register_profile("ci", max_examples=25, **_common_settings)
-settings.register_profile("dev", max_examples=10, **_common_settings)
+settings.register_profile("ci_pr", derandomize=True, max_examples=80, **_common_settings)
```
Comment (ftschindler) [inline conftest.py:19]: I don't mind the change coming with this pr, but is this related to dune-gdt?

Comment (renefritze) [reply conftest.py:19]: Yes. This stops random errors in CI.
